fluss-lake/fluss-lake-format-paimon/src/main/java/com/alibaba/fluss/lake/paimon/PaimonLakeCatalog.java

```diff
@@ -0,0 +1,177 @@
+                                        + "successfully, please try again.",
+                                tablePath, tablePath.getDatabaseName()));
+            }
+        }
+    }
+
+    private void createTable(Identifier tablePath, Schema schema)
+            throws Catalog.DatabaseNotExistException {
+        try {
+            // not ignore if table exists
+            paimonCatalog.createTable(tablePath, schema, false);
+        } catch (Catalog.TableAlreadyExistException e) {
+            throw new TableAlreadyExistException("Table " + tablePath + " already exists.");
+        }
+    }
+
+    private void createDatabase(String databaseName) {
+        try {
+            // ignore if exists
+            paimonCatalog.createDatabase(databaseName, true);
+        } catch (Catalog.DatabaseAlreadyExistException e) {
+            // do nothing, shouldn't throw since ignoreIfExists
+        }
+    }
+
+    private Identifier toPaimonIdentifier(TablePath tablePath) {
+        return Identifier.create(tablePath.getDatabaseName(), tablePath.getTableName());
+    }
+
+    private Schema toPaimonSchema(TableDescriptor tableDescriptor) {
+        Schema.Builder schemaBuilder = Schema.newBuilder();
+        Options options = new Options();
+
+        // When bucket key is undefined, it should use dynamic bucket (bucket = -1) mode.
+        List<String> bucketKeys = tableDescriptor.getBucketKeys();
+        if (!bucketKeys.isEmpty()) {
+            int numBuckets =
+                    tableDescriptor
+                            .getTableDistribution()
+                            .flatMap(TableDescriptor.TableDistribution::getBucketCount)
+                            .orElseThrow(
+                                    () ->
+                                            new IllegalArgumentException(
+                                                    "Bucket count should be set."));
+            options.set(CoreOptions.BUCKET, numBuckets);
+            options.set(CoreOptions.BUCKET_KEY, String.join(",", bucketKeys));
+        } else {
+            options.set(CoreOptions.BUCKET, CoreOptions.BUCKET.defaultValue());
+        }
+
+        Map<String, DataType> systemColumns = new LinkedHashMap<>();
+        if (!tableDescriptor.hasPrimaryKey()) {
+            // for log table, need to set bucket, offset and timestamp as system metadata columns
+            systemColumns.put(BUCKET_COLUMN_NAME, DataTypes.INT());
+            systemColumns.put(OFFSET_COLUMN_NAME, DataTypes.BIGINT());
+            // we use timestamp_ltz type
+            systemColumns.put(TIMESTAMP_COLUMN_NAME, DataTypes.TIMESTAMP_WITH_LOCAL_TIME_ZONE());
+        }
+
+        // set schema
+        for (com.alibaba.fluss.metadata.Schema.Column column :
+                tableDescriptor.getSchema().getColumns()) {
+            String columnName = column.getName();
+            if (systemColumns.containsKey(columnName)) {
+                throw new InvalidTableException(
```

Comment:
I created an issue https://github.com/alibaba/fluss/issues/810 to avoid creating table with system columns even if the table is not enabled lake.